server/src/main/java/org/candlepin/async/JobMessageDispatcher.java

```diff
@@ -138,7 +138,7 @@ private synchronized CPMProducer getProducer() throws CPMException {
      * @throws JobMessageDispatchException
      *  if the message cannot be posted for any reason
      */
-    public void postJobMessage(JobMessage jobMessage) throws JobMessageDispatchException {
+    public synchronized void postJobMessage(JobMessage jobMessage) throws JobMessageDispatchException {
```

Comment:
Since this is a restriction of Artemis, and not something we care about at this level, we should try adding the synchronized flag onto the Artemis implementations of the messaging system. In this case, ArtemisProducer.send, ArtemisSession.commit and .rollback (probably .close, too)

Reply:
Done